pkg/apis/maistra/v2/security.go

```diff
@@ -165,6 +168,14 @@ type CustomCertificateAuthorityConfig struct {
 	Address string `json:"address,omitempty"`
 }
 
+type CertManagerCertificateAuthorityConfig struct {
+	// Address is the grpc address for an Istio compatible certificate authority endpoint.
+	// .Values.global.caAddress
+	Address             string `json:"address,omitempty"`
+	PilotCertSecretName string `json:"pilotSecretName,omitempty"`
+	RootCAConfigMapName string `json:"rootCAConfigMapName,omitempty"`
```

Comment:
Would it be easier to use the default cacerts?  Is the issue that the root CA gets configured into a ConfigMap resource instead of a Secret?  Seems like if the Certificate were configured to use cacerts as the secret, everything would work without the extra volume mounts.

Reply:
The trouble is that the `cacerts` configmap is designed to load an external plugin CA root (private key and cert) which is then used by pilot to sign all future csr including it's own serving cert., this is a less secure version of what istio-csr does. I believe It's not possible to have istio-csr work using that parameter

The parameters above are needed by istio-csr, it assumes an existing external CA issuer (handled by cert-manager), but still needs to manually inject the pilot server serving cert. which has already been signed by the external CA (istio-csr handles this automatically on chart install). [This is currently the only way to install istio-csr in the upstream istio too.](https://github.com/cert-manager/istio-csr/blob/main/hack/istio-config-1.10.0.yaml)

We do not use any injected CA to sign any new secrets but just as a trust bundle anchor, every new cert when signed goes directly to our istio-csr server for signing where it's paired with a cert-manager issuer.

here's a process diagram of a CSR during execution time:

![renewal-istio-csr-certificate (1)](https://user-images.githubusercontent.com/13947576/166435442-dd53f8de-8d9b-4741-9734-a9051472286d.png)

During configuration what happens when you set cacerts is that pilot starts to generate their own CA, assuming basically that you're still using citadel for signing.
I've spent a decent amount of time trying to hack around the `cacerts` configmap along with the existing "custom" method to fit our use case to no avail.
I am pretty confident this is the only way but happy to be wrong on this!

If you still think this is wrong could you care to explain to me at a high level how each component will be configured? ie. what fields will be set in cacerts and roughly how the istio-csr config would work?

For more details on overall design options we explored: https://docs.google.com/document/d/1CwACez6pFPi_z3CFY1bB4HwIbwT3VQOz_NDUn3cQqv0/edit#